src/main/scala/com/target/data_validator/validator/ColumnBased.scala

```diff
@@ -35,26 +35,40 @@ abstract class ColumnBased(column: String, condTest: Expression) extends CheapCh
   }
 }
 
-case class MinNumRows(minNumRows: Long) extends ColumnBased("", ValidatorBase.L0) {
+case class MinNumRows(minNumRows: Json) extends ColumnBased("", ValidatorBase.L0) {
   override def name: String = "MinNumRows"
 
-  override def substituteVariables(dict: VarSubstitution): ValidatorBase = this
+  override def substituteVariables(dict: VarSubstitution): ValidatorBase = {
+    val ret = MinNumRows(getVarSubJson(minNumRows, "minNumRows", dict))
+    getEvents.foreach(ret.addEvent)
+    ret
+  }
 
   override def configCheck(df: DataFrame): Boolean = {
-    if (minNumRows <= 0) {
-      val msg = s"MinNumRows: $minNumRows <= 0"
+
+    def notNaturalNumber(): Unit = {
+      val msg = "minNumRows must be a natural number"
       logger.error(msg)
       addEvent(ValidatorError(msg))
-      failed = true
-      true
-    } else {
-      false
     }
+
+    minNumRows.asNumber match {
+      case Some(jsonNumber) => jsonNumber.toLong match {
+        case Some(x) if x > 0 =>
+        case _ => notNaturalNumber()
+      }
+      case _ => notNaturalNumber()
+    }
+    failed
   }
 
   override def quickCheck(row: Row, count: Long, idx: Int): Boolean = {
-    failed = count < minNumRows
-    val pctError = if (failed) calculatePctError(minNumRows, count) else "0.00%"
+    // Convert to `JsonNumber` then to `Long`
+    // safe because already handled in `configCheck`
+    val minNumRowsLong = minNumRows.asNumber.get.toLong.get
```

Comment:
You could define this parsing in one place as a Try[Long]/Either/etc and then pattern match that value at validation time and "unsafely" access it at check time.